instance/tests/models/test_instance.py

```diff
@@ -436,3 +577,96 @@ def update_status(self, provisioned=False, rebooting=False):
         with patch('instance.models.server.OpenStackServer.start'):
             instance.provision()
         self.assertEqual(mock_run_playbook.call_count, 1)
+
+    @patch_os_server
+    @patch('instance.models.server.openstack.create_server')
+    @patch('instance.models.server.OpenStackServer.update_status')
+    @patch('instance.models.server.OpenStackServer.sleep_until_status')
+    @patch('instance.models.server.OpenStackServer.reboot')
+    @patch('instance.models.instance.gandi.set_dns_record')
+    @patch('instance.models.instance.OpenEdXInstance.run_playbook')
+    @patch('instance.models.instance.OpenEdXInstance.provision_external_databases')
+    def test_provision_with_external_databases(self, os_server_manager, mock_provision_databases, mock_run_playbook,
+                                               mock_set_dns_record, mock_server_reboot, mock_sleep_until_status,
+                                               mock_update_status, mock_openstack_create_server):
+        """
+        Run provisioning sequence, with external databases
+        """
+        mock_openstack_create_server.return_value.id = 'test-run-provisioning-server'
+        os_server_manager.add_fixture('test-run-provisioning-server', 'openstack/api_server_2_active.json')
+
+        instance = OpenEdXInstanceFactory(sub_domain='run.provisioning', ephemeral_databases=False)
+        instance.provision()
+        self.assertEqual(mock_provision_databases.call_count, 1)
+
+    @patch('instance.models.instance.OpenEdXInstance.provision_external_mysql')
+    @patch('instance.models.instance.OpenEdXInstance.provision_external_mongo')
+    def test_provision_external_databases(self, mock_provision_mysql, mock_provision_mongo):
+        """
+        Provision mysql and mongo databases
+        """
+        instance = OpenEdXInstanceFactory()
+        instance.provision_external_databases()
+        self.assertEqual(mock_provision_mysql.call_count, 1)
+        self.assertEqual(mock_provision_mongo.call_count, 1)
+
+    def test_provision_external_mysql(self):
+        """
+        Provision external mysql database
+        """
+        instance = OpenEdXInstanceFactory()
+        instance.provision_external_mysql()
+
+        # Has the database been created?
+        databases = subprocess.check_output("mysql -u root -e 'SHOW DATABASES'", shell=True).decode()
+        self.assertIn(instance.database_name, databases)
+
+        try:
+            # Can the database user access it?
+            #pylint: disable=missing-format-attribute
```

Comment:
@omarkhan To avoid this, you could split the next line into two:

```python
mysql_cmd = 'mysql -u {user} --password={password} -e 'SHOW TABLES' {db_name}".format(
    user=instance.database_user,
    password=instance.database_pass,
    db_name=instance.database_name
)
tables = subprocess.call(mysql_cmd, shell=True)
```